File: Web/src/types/operation-log.ts
import type { PagedRequest } from './common'

export interface OperationLogQuery extends PagedRequest {
  serviceName?: string
  userId?: string | null
  traceId?: string
  method?: string
  path?: string
  operationName?: string
  success?: boolean | null
  statusCode?: number | null
  keyword?: string
  startTimeUtc?: string | null
  endTimeUtc?: string | null
}

export interface OperationLogListItem {
  id: number
  createdAtUtc: string
  serviceName: string
  userId?: string | null
  userName?: string | null
  operationName: string
  path: string
  method: string
  ip: string
  statusCode: number
  success: boolean
  durationMs: number
  traceId: string
  exceptionMessage?: string | null
}

export interface OperationLogDetail extends OperationLogListItem {
  spanId?: string | null
  requestJson?: string | null
  responseJson?: string | null
  exceptionType?: string | null
  exceptionStackTrace?: string | null
  tagsJson?: string | null
  extraJson?: string | null
}

File: Api/src/Infrastructures/ST.Infra.EntityFramework/OperationLogs/OperationLog.cs
namespace ST.Infra.EntityFramework.OperationLogs;

public sealed class OperationLog
{
	public long Id { get; set; }

	public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;

	public string ServiceName { get; set; } = string.Empty;

	public string TraceId { get; set; } = string.Empty;

	public string? SpanId { get; set; }

	public Guid? UserId { get; set; }

	public string? UserName { get; set; }

	public string OperationName { get; set; } = string.Empty;

	public string Path { get; set; } = string.Empty;

	public string Method { get; set; } = string.Empty;

	public string Ip { get; set; } = string.Empty;

	public int StatusCode { get; set; }

	public bool Success { get; set; }

	public long DurationMs { get; set; }

	public string? RequestJson { get; set; }

	public string? ResponseJson { get; set; }

	public string? ExceptionType { get; set; }

	public string? ExceptionMessage { get; set; }

	public string? ExceptionStackTrace { get; set; }

	/// <summary>
	/// 租户 ID
	/// </summary>
	public Guid? TenantId { get; set; }

	/// <summary>
	/// 业务标签（JSON），预留扩展（如业务ID、租户、模块等）。
	/// </summary>
	public string? TagsJson { get; set; }

	/// <summary>
	/// 额外扩展字段（JSON）。
	/// </summary>
	public string? ExtraJson { get; set; }
}

